utils/frame/benchmarking-cli/src/machine/hardware.rs

```diff
@@ -18,8 +18,64 @@
 //! Contains types to define hardware requirements.
 
 use lazy_static::lazy_static;
-use serde::{Deserialize, Serialize};
-use std::fmt;
+use sc_sysinfo::{Throughput, Unit};
+use serde::{
+	de::{Error, MapAccess, Visitor},
+	ser::SerializeMap,
+	Deserialize, Deserializer, Serialize, Serializer,
+};
+use sp_std::{fmt, fmt::Formatter};
+
+fn serialize_throughput<S>(t: &Throughput, serializer: S) -> Result<S::Ok, S::Error>
+where
+	S: Serializer,
+{
+	let mut map = serializer.serialize_map(Some(1))?;
+	let (value, unit) = t.normalize();
+	let unit_as_str = match unit {
+		Unit::GiBs => "GiBs",
+		Unit::MiBs => "MiBs",
+		Unit::KiBs => "KiBs",
+	};
+
+	map.serialize_entry(&unit_as_str, &value)?;
+	map.end()
```

Comment:
@bkchr I explicitly requested that this will be done the way it is now. (:

See these comments why:

https://github.com/paritytech/substrate/pull/12368#discussion_r986833804
https://github.com/paritytech/substrate/pull/12368#discussion_r988985571

Maybe a comment in the code would be good why this `match` is currently duplicated? Or just don't serialize/deserialize the units at all and keep it all in MB/s (but that's up to @ggwpez).

Reply:
Same argument as above - just the maintainability of the json file. Since having units is self explanatory, otherwise it needs docs to explain what it means.

Reply:
Hmm okay, I'm not really convinced of having the mibs etc be part of the json. However, I also don't really care. If @ggwpez is happy, he can approve and I will not complain.